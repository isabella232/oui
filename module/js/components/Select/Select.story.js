import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean as _boolean } from '@storybook/addon-knobs';
import Select from './index.js';
var stories = storiesOf('Select', module);
stories.addDecorator(withKnobs).addDecorator(function (story) {
  return React.createElement("div", {
    id: "root-preview"
  }, story());
});
stories.add('Default', function () {
  return React.createElement(Select, {
    isDisabled: _boolean('isDisabled', false),
    name: "zoo",
    id: "zoo"
  }, React.createElement("option", {
    value: "one"
  }, "This is option one"), React.createElement("option", {
    value: "two"
  }, "And this is option two"));
});