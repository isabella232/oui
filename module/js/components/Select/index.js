import React from 'react';
import PropTypes from 'prop-types';
/**
 * Generates an <select> element
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

var Select = function Select(_ref) {
  var children = _ref.children,
      id = _ref.id,
      isDisabled = _ref.isDisabled,
      name = _ref.name;
  return React.createElement("select", {
    "data-oui-component": true,
    disabled: isDisabled,
    name: name,
    id: id,
    className: "oui-select"
  }, children);
};

Select.propTypes = {
  /** options elements */
  children: PropTypes.node,

  /** select id */
  id: PropTypes.string,

  /** boolean for when select is disabled */
  isDisabled: PropTypes.bool,

  /** select name */
  name: PropTypes.string
};
Select.displayName = 'Select';
export default Select;