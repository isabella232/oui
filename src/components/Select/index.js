import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Label from '../Label';

/**
 * Generates an <select> element
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

const Select = ({
  children,
  displayError,
  id,
  isDisabled,
  isFullWidth,
  isOptional,
  isRequired,
  label,
  onChange,
  name,
  note,
  testSection,
}) => {
  const renderSelect = (
    <select
      data-oui-component={ true }
      data-test-section={ testSection }
      disabled={ isDisabled }
      required={ isRequired }
      name={ name }
      id={ id }
      className={ classNames('oui-select', { 'width--1-1': isFullWidth }) }
      onChange={ onChange }>
      {isRequired && <option value="">Please select an option</option>}
      {children}
    </select>
  );

  if (label) {
    return (
      <div
        data-oui-component={ true }
        className={ classNames({ 'oui-form-bad-news': displayError }) }>
        <Label isRequired={ isRequired } isOptional={ isOptional } inputId={ id }>
          {label}
        </Label>
        {renderSelect}
        {note && (
          <div
            className="oui-form-note"
            data-test-section={ testSection && testSection + '-note' }>
            {note}
          </div>
        )}
      </div>
    );
  }
  return renderSelect;
};

Select.propTypes = {
  /** options elements */
  children: PropTypes.node,
  /** Whether or not this Select should be in an error state */
  displayError: PropTypes.bool,
  /** Select ID */
  id: PropTypes.string,
  /** Boolean for when Select is disabled */
  isDisabled: PropTypes.bool,
  /** Boolean to make Select take up 100% width of container */
  isFullWidth: PropTypes.bool,
  /** Boolean to indicate this Select is optional */
  isOptional: PropTypes.bool,
  /** Boolean to indicate this Select is required */
  isRequired: PropTypes.bool,
  /** The label for the Select */
  label: PropTypes.string,
  /** Select name */
  name: PropTypes.string,
  /** Message to show underneath the Select */
  note: PropTypes.string,
  /** Function that fires when value of the select changes */
  onChange: PropTypes.func,
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
};

Select.displayName = 'Select';

export default Select;
