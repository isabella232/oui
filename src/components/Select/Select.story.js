import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, boolean } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';

import Select from './index.js';

const stories = storiesOf('Select', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(withInfo)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories.add('default', (() => {
  return (
    <Select isDisabled={ boolean('isDisabled', false) } name='zoo' id='zoo' >
      <option value="one">This is option one</option>
      <option value="two">And this is option two</option>
    </Select>
  );
}));
