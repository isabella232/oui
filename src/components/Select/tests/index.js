import React from 'react';
import Select from '../index';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('components/Select', () => {
  it('should render correctly when declared with no properties', () => {
    const output = shallow(
      <Select>
        <option value="one">This is option one</option>
        <option value="two">And this is option two</option>
      </Select>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly when isDisabled prop is declared', () => {
    const output = shallow(
      <Select isDisabled={ true }>
        <option value="one">This is option one</option>
        <option value="two">And this is option two</option>
      </Select>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly when a label is supplied', () => {
    const output = shallow(
      <Select isDisabled={ false } label="Test Select">
        <option value="one">This is option one</option>
        <option value="two">And this is option two</option>
      </Select>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly when isFullWidth is declared', () => {
    const output = shallow(
      <Select isDisabled={ false } label="Test Select" isFullWidth={ true }>
        <option value="one">This is option one</option>
        <option value="two">And this is option two</option>
      </Select>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly when displayError is declared and a note is present', () => {
    const output = shallow(
      <Select
        isDisabled={ false }
        label="Test Select"
        note="This field is required"
        displayError={ true }>
        <option value="one">This is option one</option>
        <option value="two">And this is option two</option>
      </Select>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should call the onChange function passed in', () => {
    const handler = {
      onChange: event => {},
    };
    spyOn(handler, 'onChange');

    const component = mount(
      <Select onChange={ handler.onChange }>
        <option value="one">This is option one</option>
        <option value="two">And this is option two</option>
      </Select>
    );

    component.simulate('change');

    expect(handler.onChange).toHaveBeenCalled();
  });
});
