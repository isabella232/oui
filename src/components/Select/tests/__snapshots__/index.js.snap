// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Select should render correctly when a label is supplied 1`] = `
<div
  className=""
  data-oui-component={true}
>
  <Label>
    Test Select
  </Label>
  <select
    className="oui-select"
    data-oui-component={true}
    disabled={false}
  >
    <option
      value="one"
    >
      This is option one
    </option>
    <option
      value="two"
    >
      And this is option two
    </option>
  </select>
</div>
`;

exports[`components/Select should render correctly when declared with no properties 1`] = `
<select
  className="oui-select"
  data-oui-component={true}
>
  <option
    value="one"
  >
    This is option one
  </option>
  <option
    value="two"
  >
    And this is option two
  </option>
</select>
`;

exports[`components/Select should render correctly when displayError is declared and a note is present 1`] = `
<div
  className="oui-form-bad-news"
  data-oui-component={true}
>
  <Label>
    Test Select
  </Label>
  <select
    className="oui-select"
    data-oui-component={true}
    disabled={false}
  >
    <option
      value="one"
    >
      This is option one
    </option>
    <option
      value="two"
    >
      And this is option two
    </option>
  </select>
  <div
    className="oui-form-note"
  >
    This field is required
  </div>
</div>
`;

exports[`components/Select should render correctly when isDisabled prop is declared 1`] = `
<select
  className="oui-select"
  data-oui-component={true}
  disabled={true}
>
  <option
    value="one"
  >
    This is option one
  </option>
  <option
    value="two"
  >
    And this is option two
  </option>
</select>
`;

exports[`components/Select should render correctly when isFullWidth is declared 1`] = `
<div
  className=""
  data-oui-component={true}
>
  <Label>
    Test Select
  </Label>
  <select
    className="oui-select width--1-1"
    data-oui-component={true}
    disabled={false}
  >
    <option
      value="one"
    >
      This is option one
    </option>
    <option
      value="two"
    >
      And this is option two
    </option>
  </select>
</div>
`;
